packages/smart-wallet/src/walletFactory.js

```diff
@@ -274,16 +278,10 @@ export const prepare = async (zcf, privateArgs, baggage) => {
           return wallet;
         };
 
-        const finisher = walletReviver
-          ? async (_address, _wallet) => {
-              const isRevive = await E(walletReviver).ackWallet(address);
-              isNew = !isRevive;
-            }
-          : undefined;
-
-        return provider
-          .provideAsync(address, maker, finisher)
-          .then(w => [w, isNew]);
+        const wallet = await provider.provideAsync(address, maker);
+        return isNew && walletReviver
+          ? E(walletReviver).ackWallet(address, wallet)
```

Comment:
This relies on the reviver to pass the wallet through without changing it. Does this component already rely on the reviver? I suppose it does in the case of a revived wallet.

Perhaps worth a comment?